app/controllers/HomeController.scala

```diff
@@ -0,0 +1,94 @@
+package controllers
+
+import javax.inject._
+
+import actors._
+import akka.actor._
+import akka.event.{Logging, LoggingAdapter}
+import akka.stream.scaladsl._
+import akka.stream._
+import play.api.libs.json._
+import play.api.mvc._
+import akka.pattern.ask
+import akka.util.Timeout
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class HomeController @Inject()(@Named("stocksActor") stocksActor: ActorRef,
+                               @Named("userParentActor") userParentActor: ActorRef)
+                              (implicit actorSystem: ActorSystem,
+                               mat: Materializer,
+                               ec: ExecutionContext) extends Controller {
+
+  // Use a direct reference to SLF4J
+  private val logger = org.slf4j.LoggerFactory.getLogger("controllers.HomeController")
+
+  // Home page that renders template
+  def index = Action { implicit request =>
+    Ok(views.html.index())
+  }
+
+  // Open up a websocket using JsValue in and out.
+  def ws = WebSocket.accept[JsValue, JsValue] { request =>
+
+    // create an actor ref source and associated publisher for sink
+    val (outActor, publisher) = {
+      val source = {
+        // http://doc.akka.io/docs/akka/current/scala/stream/stream-flows-and-basics.html#stream-materialization
+        // http://doc.akka.io/docs/akka/current/scala/stream/stream-integrations.html#integrating-with-actors
+
+        // If you want to log on a flow, you have to use a logging adapter.
+        // http://doc.akka.io/docs/akka/2.4.4/scala/logging.html#SLF4J
+        val logging = Logging(actorSystem.eventStream, logger.getName)
+        Source.actorRef[JsValue](10, OverflowStrategy.dropTail)
+          .log("actorRefSource")(logging)
+      }
+      source.toMat(Sink.asPublisher(fanout = false))(Keep.both).run()
+    }
+
+    // Create a user actor off the request id and attach it to the source
+    val id = request.id.toString
+
+    // Use assisted injection to configure the child actor.
+    val userActorFuture = {
+      implicit val timeout = Timeout(100.millis)
+      (userParentActor ? UserParentActor.Create(id, outActor)).mapTo[ActorRef]
+    }
+
+    // Using await isn't great here.
+    val awaitDuration = 100.millis
+    val userActor = scala.concurrent.Await.result(userActorFuture, awaitDuration)
```

Comment:
I actually found a similar situation, when people were not able to create the Actor because needed to pass in dependencies which were resolved in the future... 

I have a dream... of providing a proxy that'd buffer messages until the actor is created and then become a direct forwarder to it. Would be simple to implement actually - actor with this one as future inside, and becomes "forwarding" once the future completes